mlflow/models/evaluation/base.py

```diff
@@ -617,6 +617,11 @@ def __init__(
                 error_code=INVALID_PARAMETER_VALUE,
             )
 
+        if self._has_predictions:
+            _validate_dataset_type_supports_predictions(
+                data=data, supported_predictions_dataset_types=self._supported_dataframe_types
+            )
```

Comment:
We removed the upstream assertion that `predictions` must be a pandas dataframe when `model` is unspecified: https://github.com/mlflow/mlflow/pull/11831/files#r1580245904. 

We need to perform this check before extracting the `predictions` column from the `EvaluationDataset`, else we might run into a nasty "can't extract predictions from numpy array" error. Here, we verify that the data is a dataframe.